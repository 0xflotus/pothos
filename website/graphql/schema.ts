import { docsIndex } from '../util/build-index.mjs';
import { builder } from './builder';
import {
  DocsIndex,
  DocsSection,
  SearchMatch,
  SearchMatcher,
  SearchMatchKind,
  SearchResult,
} from './search-matcher';

const docsIndexFiles = docsIndex as DocsIndex[];

const DocsPage = builder.objectRef<DocsIndex>('DocsPage').implement({
  fields: (t) => ({
    title: t.exposeString('name'),
    description: t.exposeString('description'),
    link: t.exposeString('link'),
    md: t.exposeString('md'),
    html: t.exposeString('html'),
  }),
});

builder.enumType(SearchMatchKind, { name: 'SearchMatchKind' });

const SearchMatchRef = builder.objectRef<SearchMatch>('SearchMatch').implement({
  fields: (t) => ({
    kind: t.expose('kind', { type: SearchMatchKind }),
    section: t.expose('section', { type: DocsSectionRef, nullable: true }),
  }),
});

const SearchResultRef = builder.objectRef<SearchResult>('SearchResult').implement({
  fields: (t) => ({
    matches: t.expose('matches', { type: [SearchMatchRef] }),
    doc: t.expose('doc', { type: DocsPage, nullable: true }),
  }),
});

const DocsSectionRef = builder.objectRef<DocsSection>('DocsSection').implement({
  fields: (t) => ({
    heading: t.exposeString('heading'),
    link: t.exposeString('link'),
    md: t.exposeString('md'),
    html: t.exposeString('html'),
  }),
});

builder.queryType({
  fields: (t) => ({
    search: t.field({
      type: [SearchResultRef],
      args: {
        query: t.arg.string({
          required: true,
        }),
      },
      resolve: (_, { query }) => new SearchMatcher(query).matchDocs(docsIndexFiles).getResults(),
    }),
  }),
});

export const schema = builder.toSchema({});
