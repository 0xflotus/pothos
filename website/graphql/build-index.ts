/* eslint-disable unicorn/no-process-exit */
/* eslint-disable no-process-exit */
/* eslint-disable @typescript-eslint/no-implicit-any-catch */
// import algoliasearch from 'algoliasearch/lite';
import dotenv from 'dotenv';

try {
  dotenv.config();

  if (!process.env.NEXT_PUBLIC_ALGOLIA_APP_ID) {
    throw new Error('NEXT_PUBLIC_ALGOLIA_APP_ID is not defined');
  }

  if (!process.env.ALGOLIA_SEARCH_ADMIN_KEY) {
    throw new Error('ALGOLIA_SEARCH_ADMIN_KEY is not defined');
  }
} catch (error) {
  console.error(error);
  process.exit(1);
}

// const searchFormat = loadedFiles.map((file) => {
//   if (!(file.data.title as string)) {
//     throw new Error(`Missing title for ${file.path}`);
//   }

//   return {
//     objectID: file.path,
//     title: file.data.title as string,
//     description: file.data.description as string,
//     slug: (file.data.slug as string) || toSlug(file.path),
//     tagsCollection: { tags: file.data.tags as string[] },
//     date: file.data.date as string,
//     type: 'docs',
//     content: file.content,
//   };
// });
