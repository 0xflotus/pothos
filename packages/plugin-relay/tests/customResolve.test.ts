import { execute } from 'graphql';
import { gql } from 'graphql-tag';
import schema from './examples/custom-resolve/schema';

describe('custom node resolve', () => {
  it('resolves a node with a global id', async () => {
    const query = gql`
      query {
        node(id: "TnVtYmVyOjI=") {
          ... on BatchNumber {
            batchNumber: number
          }
          ... on Number {
            number
          }
        }
      }
    `;

    const result = await execute({
      schema,
      document: query,
      contextValue: {},
    });

    expect(result).toMatchSnapshot();
  });
});
