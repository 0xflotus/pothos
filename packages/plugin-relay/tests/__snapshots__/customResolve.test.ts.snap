// Vitest Snapshot v1

exports[`cursor based connection > queries 1`] = `
{
  "data": {
    "node": {
      "number": 2,
    },
  },
}
`;
